### PR TITLE
Create a generic and default load_processes() function

### PR DESCRIPTION
**Pull request by bradjc**
Historically the `load_process()` function has been in the board's main.rs file. This has lead to some duplicated code. This PR adds a generic version to process.rs that platforms can use if they wish.

It also updates the imix and hail boards to use the generic version.

### CODE REVIEW

File: kernel/src/process.rs [hunk 1]
```diff
@@ -39,6 +39,47 @@ extern "C" {
 
 pub static mut PROCS: &'static mut [Option<Process<'static>>] = &mut [];
 
+/// Helper function to load processes from flash into an array of active
+/// processes. This is the default template for loading processes, but a board
+/// is able to create its own `load_processes()` function and use that instead.
+pub unsafe fn load_processes(num_procs: usize,
+    app_memory: &mut [u8],
+    procs: &'static mut [Option<Process<'static>>],
```
Comment (alevy) [inline kernel/src/process.rs]: this doesn't have to be `'static`

File: kernel/src/process.rs [hunk 2]
```diff
@@ -39,6 +39,47 @@ extern "C" {
...
+    fault_response: FaultResponse) -> &'static mut [Option<Process<'static>>] {
```
Comment (alevy) [inline kernel/src/process.rs]: why return anything? isn't just the same as `procs`? I believe returning something here technically results in an alias, which is bad...

File: kernel/src/process.rs [hunk 3]
```diff
@@ -39,6 +39,47 @@ extern "C" {
...
+    extern "C" {
+        /// Beginning of the ROM region containing app images.
+        static _sapps: u8;
```
Comment (alevy) [inline kernel/src/process.rs]: I think it might be better to have the caller pass in this value instead

**Comment by bradjc** [comment]
Updated.

**Comment by bradjc** [comment]
Passing in a `num_procs` didn't make much sense since the `procs` array has a length. The parameter has been removed.

**Pull request merged**